src/backends/beamcoder.ts

```diff
@@ -365,6 +367,47 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
         return this.getFrameAtPts(targetPts);
     }
 
+    /**
+     * Dump one frame's image data (for canvas) at a specific time
+     *
+     * This method can seek as required, but generally, it is designed to be
+     * performant in the cases where we progressively read a video frame by frame.
+     *
+     * So the implementation in here should not seek all the time, but rather
+     * read packets as they come, when possible,
+     *
+     * @see getFrameAtPts()
+     */
+    async getFrameImageDataAtTime(targetTime: number): Promise<ImageData> {
+        LOG_SINGLE_FRAME_DUMP_FLOW && console.log(`Requesting to dump a frame at Time(s)=${targetTime}`);
+        const targetPts = Math.floor(this.timeToPts(targetTime));
+        const frame = await this.getFrameAtPts(targetPts);
+
+        const components = 4; // 4 components: r, g, b and a
+        const size = frame.width * frame.height * components;
+        const rawData = new Uint8ClampedArray(size);
+        const sourceLineSize = frame.linesize as unknown as number;
+        const pixels = frame.data[0] as Uint8Array;
+
+        // libav creates larger buffers because it makes their internal code simpler.
+        // we have to trim a part at the right of each pixel row.
+        for (let i = 0; i < frame.height; i++) {
```

Comment:
Wow, we still need to go through every line and adjust the width!? There isn't a filter you can use for this @antoineMoPa?

Reply:
@animanathome  We could encode + mux as raw image format (using image2), but this would take a bit more time to develop and I'm not sure it would be faster. I see this PR as a first step to clean up the API and later we can optimize performance.

Performance is not so bad either! For a hd frame extraction, on my machine, it's 1.7ms to 2.5ms. By comparison, `getFrameAtPts` takes anywhere from 2ms to 13ms, sometimes more.

Reply:
Sounds good